src/tree/SubscriptionTreeItem.ts

```diff
@@ -60,34 +68,80 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
-        context.showCreatingTreeItem(accountName);
-        await wizard.execute();
-        // don't wait
-        vscode.window.showInformationMessage(`Successfully created account "${accountName}".`);
-        return await this.initCosmosDBChild(client, nonNullProp(wizardContext, 'databaseAccount'));
+        await wizardInitial.prompt();
+
+        if (wizardContextInitial.accountType === "cosmosdb") {
+            const client: CosmosDBManagementClient = createAzureClient(this.root, CosmosDBManagementClient);
+            const wizardContext: ICosmosDBWizardContext = Object.assign(context, this.root);
+
+            promptSteps.push(
+                new CosmosDBAccountNameStep(),
+                new CosmosDBAccountApiStep(),
+                new ResourceGroupListStep()
+            );
+            LocationListStep.addStep(wizardContext, promptSteps);
+
+            const wizard = new AzureWizard(wizardContext, {
+                promptSteps,
+                executeSteps: [
+                    new CosmosDBAccountCreateStep()
+                ],
+                title: 'Create new Cosmos DB account'
+            });
+
+            await wizard.prompt();
+
+            wizardContext.telemetry.properties.defaultExperience = wizardContext.defaultExperience?.api;
+
+            const accountName: string = nonNullProp(wizardContext, 'accountName');
+            context.showCreatingTreeItem(accountName);
+            await wizard.execute();
+            // don't wait
+            vscode.window.showInformationMessage(`Successfully created account "${accountName}".`);
+            return await this.initCosmosDBChild(client, nonNullProp(wizardContext, 'databaseAccount'));
+        } else {
+            const wizardContext: IPostgresWizardContext = Object.assign(context, this.root);
+            promptSteps.push(
+                new PostgresServerNameStep(),
+                new ResourceGroupListStep(),
+                new PostgresServerCredStep(),
+                new PostgresServerFirewallStep()
+            );
+            LocationListStep.addStep(wizardContext, promptSteps);
+            const wizard = new AzureWizard(wizardContext, {
+                promptSteps,
+                executeSteps: [
+                    new PostgresServerCreateStep()
+                ],
+                title: 'Create new Postgres account'
+            });
+            await wizard.prompt();
+            const serverName: string = nonNullProp(wizardContext, 'serverName');
+            context.showCreatingTreeItem(serverName);
+            await wizard.execute();
+
+            vscode.window.showInformationMessage(`Successfully created account "${wizardContext.serverName}".`);
+            const serverTree: PostgresServerTreeItem = new PostgresServerTreeItem(this, nonNullProp(wizardContext, 'server'));
+            let user: string = nonNullProp(wizardContext, 'adminUser');
+            const usernameSuffix: string = `@${serverName}`;
```

Comment:
nit: There's a bit of overlap here with [enterPostgresCredentials.ts](https://github.com/microsoft/vscode-cosmosdb/blob/master/src/postgres/commands/enterPostgresCredentials.ts#L21). It could probably be combined into a function called "validateUsername" or something along those lines

Reply:
The database login part doesn't seem to need extensive username validation compared to the create step. 
Do you mean combining just to simplify the username section in SubscriptionTreeItem or combining enterPostgresCredentials.ts with PostgresServerCredStep...?

Reply:
I was talking about [these lines](https://github.com/microsoft/vscode-cosmosdb/blob/6c8f5d8e3a670efffb207f8cf422950301d5cd31/src/tree/SubscriptionTreeItem.ts#L137-L140) being duplicated but its probs fine as it is.

Reply:
Oh ok, for these lines I feel like we might have to change the enterPostgresCredentials code, I feel like that it's unnecessary given that the rest of the function is not useful in this scenario.